packages/flutter/test/widgets/text_selection_test.dart

```diff
@@ -2008,14 +2008,12 @@ class TextSelectionControlsSpy extends TextSelectionControls {
 
   @override
   Widget buildHandle(BuildContext context, TextSelectionHandleType type, double textLineHeight, [VoidCallback? onTap]) {
-    switch (type) {
-      case TextSelectionHandleType.left:
-        return ElevatedButton(onPressed: onTap, child: Text('height ${textLineHeight.toInt()}', key: leftHandleKey));
-      case TextSelectionHandleType.right:
-        return ElevatedButton(onPressed: onTap, child: Text('height ${textLineHeight.toInt()}', key: rightHandleKey));
-      case TextSelectionHandleType.collapsed:
-        return ElevatedButton(onPressed: onTap, child: Text('height ${textLineHeight.toInt()}', key: collapsedHandleKey));
-    }
+    final UniqueKey key = switch (type) {
+      TextSelectionHandleType.left      => leftHandleKey,
+      TextSelectionHandleType.right     => rightHandleKey,
+      TextSelectionHandleType.collapsed => collapsedHandleKey,
+    };
+    return ElevatedButton(onPressed: onTap, child: Text('height ${textLineHeight.toInt()}', key: key));
```

Comment:
since you're in here, maybe format this a little nicer:

```suggestion
    return ElevatedButton(
      onPressed: onTap, 
      child: Text(
        key: key
        'height ${textLineHeight.toInt()}',
      ),
    );
```